test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs

```diff
@@ -2818,7 +2818,19 @@ public void FilterWithInOperationWithDateTimeOffsetCollectionContainingSingleNul
             Assert.Equal(new object[] {null},
                 Assert.IsType<CollectionConstantNode>(inNode.Right).Collection.Select(x => x.Value));
         }
-#endregion
+
+        [Fact]
+        public void FilterWithInOperationWithOpenTypes()

```

Comment:
>FilterWithInOperationWithOpenTypes [](http://example.com/codeflow?start=20&length=34)

Do we have any end-to-end tests that shows this actually generates the appropriate LINQ expression?